src/main/java/engineering/swat/watch/impl/jdk/JDKRecursiveDirectoryWatch.java

```diff
@@ -85,27 +73,24 @@ private void handleCreate(WatchEvent ev) {
         // create till after the processing is done, so we schedule it in the background
         var fullPath = ev.calculateFullPath();
         if (!activeWatches.containsKey(fullPath)) {
-            CompletableFuture
-                .completedFuture(fullPath)
-                .thenApplyAsync(this::registerForNewDirectory, exec)
-                .thenAcceptAsync(this::publishExtraEvents, exec)
-                .exceptionally(ex -> {
-                    logger.error("Could not locate new sub directories for: {}", ev.calculateFullPath(), ex);
-                    return null;
-                });
+            try {
+                if (Files.isDirectory(fullPath)) {
+                    addNewDirectory(fullPath);
+                    reportOverflow(fullPath);
+                }
+            } catch (IOException ex) {
+                logger.error("Could not locate new sub directories for: {}", ev.calculateFullPath(), ex);
+            }
         }
     }
 
     private void handleOverflow(WatchEvent ev) {
-        logger.info("Overflow detected, rescanning to find missed entries in {}", path);
-        CompletableFuture
-            .completedFuture(ev.calculateFullPath())
-            .thenApplyAsync(this::syncAfterOverflow, exec)
-            .thenAcceptAsync(this::publishExtraEvents, exec)
-            .exceptionally(ex -> {
-                logger.error("Could not register new watch for: {} ({})", ev.calculateFullPath(), ex);
-                return null;
-            });
+        var fullPath = ev.calculateFullPath();
+        try (var children = Files.find(fullPath, 1, (p, attrs) -> p != fullPath && attrs.isDirectory())) {
```

Comment:
here the same, what if a user has configured to ignore overflow events? Does that mean we would then miss all nested directories in our watcher?